static/js/actions/email.js

```diff
@@ -33,13 +39,30 @@ export function sendSearchResultMail(
   searchRequest: Object
 ): Dispatcher<EmailSendResponse> {
   return (dispatch: Dispatch) => {
-    dispatch(initiateSendEmail());
+    dispatch(initiateSendEmail(SEARCH_EMAIL_TYPE));
     return api.sendSearchResultMail(subject, body, searchRequest).
       then(response => {
-        dispatch(sendEmailSuccess());
+        dispatch(sendEmailSuccess(SEARCH_EMAIL_TYPE));
         return Promise.resolve(response);
       }).catch(error => {
-        dispatch(sendEmailFailure(error));
+        dispatch(sendEmailFailure({type: SEARCH_EMAIL_TYPE, error: error}));
+      });
+  };
+}
+
+export function sendCourseTeamMail(
+  subject: string,
+  body: string,
+  courseId: number
+): Dispatcher<EmailSendResponse> {
+  return (dispatch: Dispatch) => {
+    dispatch(initiateSendEmail(COURSE_EMAIL_TYPE));
+    return api.sendCourseTeamMail(subject, body, courseId).
+      then(response => {
+        dispatch(sendEmailSuccess(COURSE_EMAIL_TYPE));
+        return Promise.resolve(response);
+      }).catch(error => {
+        dispatch(sendEmailFailure({type: COURSE_EMAIL_TYPE, error: error}));
```

Comment:
i definitely agree with this, but we are currently in merge-by-any-means-necessary mode with this feature. I already have a general plan for addressing this, but fixing the js tests will add more work than is sensible given the circumstances. i'll do that refactor as a separate issue or as part of #2284 depending on how the review for that PR shapes up